beacon-chain/utils/clock.go

```diff
@@ -16,3 +19,19 @@ type RealClock struct{}
 func (RealClock) Now() time.Time {
 	return time.Now()
 }
+
+// CurrentBeaconSlot based on the seconds since genesis.
+func CurrentBeaconSlot() uint64 {
+	secondsSinceGenesis := time.Since(params.GenesisTime).Seconds()
+	return uint64(math.Floor(secondsSinceGenesis / 8.0))
+}
+
+// WaitUntilTimestamp sleeps until a specific time is reached after
+// a certain duration. For example, if the genesis block
+// was at 12:00:00PM and the current time is 12:00:03PM,
+// we want the next slot to tick at 12:00:08PM so we can use
+// this helper method to achieve that purpose.
+func WaitUntilTimestamp(duration time.Duration) {
```

Comment:
I don't quite like the name of this method, but I can't think of a better alternative.